core/src/main/scala/kafka/raft/TimingWheelExpirationService.scala

```diff
@@ -17,17 +17,18 @@
 package kafka.raft
 
 import java.util.concurrent.CompletableFuture
-import kafka.utils.timer.{Timer, TimerTask}
 import org.apache.kafka.common.errors.TimeoutException
 import org.apache.kafka.raft.ExpirationService
 import org.apache.kafka.server.util.ShutdownableThread
+import org.apache.kafka.server.util.timer.{Timer, TimerTask}
 
 object TimingWheelExpirationService {
   private val WorkTimeoutMs: Long = 200L
 
-  class TimerTaskCompletableFuture[T](override val delayMs: Long) extends CompletableFuture[T] with TimerTask {
```

Comment:
why did we remove the inheritance from CompletableFuture?

Reply:
`TimerTask` is an abstract class in Java vs a trait in Scala and we can't extend two classes. Therefore, I have refactored this piece to only extend `TimerTask`.